public/pages/workflows/workflow_list/resource_list.tsx

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState, useEffect, useCallback } from 'react';
+import { columns } from '../../workflow_detail/tools/resources/columns';
+import {
+  EuiCodeBlock,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiBasicTable,
+  EuiButtonIcon,
+  RIGHT_ALIGNMENT,
+  EuiText,
+  Direction,
+} from '@elastic/eui';
+import {
+  Workflow,
+  WorkflowResource,
+  customStringify,
+} from '../../../../common';
+import {
+  getIndex,
+  getIngestPipeline,
+  getSearchPipeline,
+  useAppDispatch,
+} from '../../../store';
+import { getDataSourceId } from '../../../utils';
+
+interface ResourceListProps {
+  workflow?: Workflow;
+}
+
+/**
+ * The searchable list of resources for a particular workflow.
+ */
+export function ResourceList(props: ResourceListProps) {
+  const [allResources, setAllResources] = useState<WorkflowResource[]>([]);
+  const dispatch = useAppDispatch();
+  const dataSourceId = getDataSourceId();
+  const [itemIdToExpandedRowMap, setItemIdToExpandedRowMap] = useState<{
+    [key: string]: React.ReactNode;
+  }>({});
```

Comment:
This is inefficient for persisting the data - we should separate the data from the rendered components to save space and to simplify the data model design. I'd suggest having a map mapping ID -> string, instead of ID -> ReactNode. The `render` fn on line 160 can then take the exposed item ID, and render the ReactNode (in this case, the `EuiCodeBlock`) using the map's string value directly.

This way, we only persist the minimal state required, and let the components handle the dynamic rendering of the data.

Reply:
Update: above may be irrelevant after addressing https://github.com/opensearch-project/dashboards-flow-framework/pull/462#discussion_r1828266467

Reply:
Update 2: I see this map is required for using [expanded rows](https://oui.opensearch.org/1.6/#/tabular-content/tables#expanding-rows), so I'm fine with using that. But still prefer to fetch all of the data from redux store, and create/update the map based on the data from that store instead of the current mechanism of fetching and parsing the API response directly.

Reply:
The results from search index are not persisted because they are transient/runtime values, only needed at the particular time of execution. Redux store is for persisting/caching backend resources / data that is used across the UI, such that it is not needed to be fetched constantly. Note how many of the resource fetching of current backend resources (workflows, indices, ML models etc.) are fetched in the top-level components once to populate the redux store. Then, the downstream child components simply fetch from that store, without having to rerun the APIs and increase traffic to the cluster.

Ingest and search pipelines are persistent resources, and should be fetched in the same way. Then, the tables that leverage them don't need to make extra API calls. Additionally, the resources are then persisted in a standard, defined way. Note how we have defined interfaces for Workflows, Models, etc. that are formatted and saved in the redux store. Then, when consumed by other components, it is entirely typesafe and the data model is entirely known.

Can you explain what you mean by `Fetching all Resources Details and storing in redux Store doesn't seem optimal`? I'd consider this a blocker still for this PR, since it's directly introducing tech debt by the current implementation.

Reply:
- Tyler, As discussed I am working on **Fetching all resource details** (meaning details of all resources)present in a workflow and having them in store while rendering resourceList table . 
- Then from store will fetch details of any resource instead of calling APIs directly. 



Few Insights as per our discussion :

1. Advantages - 

- Uniformity across plugin to fetch to store and use it from there.
- When user looks at same resource details multiple times we are not doing API calls several times.  

2. Disadvantages -

- Though the user is not looking for resource details in the drill down and just opened resource list we will do API calls to get all the resources Details in that workflow into Store.